File: ruby/MobSmsClient.rb
=begin
Desc:短信http接口的ruby代码调用示例
author zxc
date 2017-03-09
=end

require 'net/http'
require 'uri'
params = {}


#修改为您的appkey.在官网（http://www.mob.com)登录后获取,接口地址联系客服
appkey = 'xxxxxxxxxxxxxxxxxxxxxxxxxxxx'
#修改为您要发送的手机号码
phone = '18912386146'
#修改为您要发送的国家代码
zone = '86'

#Web-Api验证码发送接口
get_webapi_send_uri = URI.parse('https://webapi.sms.mob.com/xxxxx')
#Web-Api验证码验证接口
get_webapi_check_uri = URI.parse('https://webapi.sms.mob.com/xxxxx')

#请求下发短信验证码
params['appkey'] = appkey
params['phone'] = phone
params['zone'] = zone
response =  Net::HTTP.post_form(get_webapi_sendmsg_uri,params)
print response.body + "\n"

#请求校验短信验证码
code = '1234'
params['appkey'] = appkey
params['phone'] = phone
params['zone'] = zone
params['code'] = code
response = Net::HTTP.post_form(get_webapi_checkcode_uri,params)
print response.body + "\n"

File: go/MobSmsClient.go
// Package: default
// File: MobSmsClient.go
// Created by: zxc(zhangxiongcai337@gmail.com)
// DATE: 17-4-17 18:01
package main
import (
	"net/http"
	"io/ioutil"
	"net/url"
    "fmt"
 )
// bingone
func main(){

	// 修改为您的appkey.在官网（http://www.mob.com)登录后获取,接口地址联系客服
	apikey 		:= "xxxxxxxxxxxxxxxxxx"
	// 修改为您要发送的手机号码
	phone 		:= "18912386146"
	// 国家代码
	zone 		:= "86"
	// 手机收到的短信验证码code
	code		:= "1234"
	
	// Web-Api验证码发送接口
	url_webapi_send_uri	  := "https://xxxxxxxxxxxxxxxxxx";
	// Web-Api验证码校验接口
	url_webapi_check_uri  := "https://xxxxxxxxxxxxxxxxxx";
	// 短信服务端验证接口
	url_webapi_verify_uri := "https://xxxxxxxxxxxxxxxxxx";


	data_webapi_send_uri	:= url.Values{"appkey": {apikey}, "phone": {phone},"zone":{zone}}
	data_webapi_check_uri	:= url.Values{"appkey": {apikey}, "phone": {phone},"zone":{zone},"code":{code}}
	data_webapi_verify_uri	:= url.Values{"appkey": {apikey}, "phone": {phone},"zone":{zone},"code":{code}}


	httpsPostForm(url_webapi_send_uri,data_webapi_send_uri)
	httpsPostForm(url_webapi_check_uri,data_webapi_check_uri)
	httpsPostForm(url_webapi_verify_uri,data_webapi_verify_uri)
}

func httpsPostForm(url string,data url.Values) {
	resp, err := http.PostForm(url,data)

	if err != nil {
		// handle error
	}

	defer resp.Body.Close()
	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		// handle error
	}

	fmt.Println(string(body))

}
